collectors/debugfs.plugin/debugfs_zswap.c

```diff
@@ -0,0 +1,353 @@
+                    "SET %s = %lld\n"
+                    "END\n",
+            metric->chart_id, metric->dimension, metric->value);
+    fflush(stdout);
+}
+
+void zswap_reject_chart(int update_every, const char *name) {
+    int i;
+    struct netdata_zswap_metric *metric = &zswap_rejected_metrics[NETDATA_ZSWAP_REJECTED_CHART];
+    if (unlikely(!metric->chart_created)) {
+        metric->chart_created = CONFIG_BOOLEAN_YES;
+        zswap_send_chart(metric, update_every, name, NULL);
+        for (i = NETDATA_ZSWAP_REJECTED_COMPRESS_POOR; zswap_rejected_metrics[i].filename; i++) {
+            metric = &zswap_rejected_metrics[i];
+            if (likely(metric->enabled))
+                fprintf(stdout,
+                    "DIMENSION '%s' '%s' %s 1 1 ''\n",
+                    metric->dimension,
+                    metric->dimension,
+                    debugfs_rrd_algorithm_name(metric->algorithm));
+        }
+    }
+
+    fprintf(stdout, "BEGIN system.zswap_%s\n", zswap_rejected_metrics[NETDATA_ZSWAP_REJECTED_CHART].chart_id);
+    for (i = NETDATA_ZSWAP_REJECTED_COMPRESS_POOR; zswap_rejected_metrics[i].filename; i++) {
+        metric = &zswap_rejected_metrics[i];
+        fprintf(stdout,
+                "SET %s = %lld\n",
+                metric->dimension,
+                metric->value);
+    }
+    fprintf(stdout, "END\n");
+    fflush(stdout);
+}
+
+int debugfs_zswap(int update_every, const char *name) {
+    int i;
+    struct netdata_zswap_metric *metric;
+    int independent_disabled = 0;
+    for (i = 0; zswap_independent_metrics[i].filename; i++) {
+        metric = &zswap_independent_metrics[i];
+        if (likely(metric->enabled)) {
+            metric->enabled = !zswap_collect_data(metric);
+            if (unlikely(metric->obsolete == CONFIG_BOOLEAN_NO))
+                zswap_independent_chart(metric, update_every, name);
+            else if (likely(metric->chart_created && metric->obsolete)) {
+                independent_disabled++;
+                zswap_send_chart(metric, update_every, name, "obsolete");
+            }
+        }
+    }
+
+    // We set as 1, because the first structure with dimensions is not zero.
+    int rejected_disabled = 1;
+    for (i = NETDATA_ZSWAP_REJECTED_COMPRESS_POOR; zswap_rejected_metrics[i].filename; i++) {
+        metric = &zswap_rejected_metrics[i];
+        metric->enabled = !zswap_collect_data(metric);
+        if (unlikely(!metric->enabled)) {
+            rejected_disabled++;
+        }
+    }
+
+    // When more than half of metrics are present we still plot chart.
+    if (unlikely(rejected_disabled < NETDATA_ZSWAP_REJECTED_END))
+        zswap_reject_chart(update_every, name);
+    else {
+        zswap_send_chart(&zswap_rejected_metrics[NETDATA_ZSWAP_REJECTED_CHART], update_every, name, "obsolete");
+        // We do not have metrics, so we are disabling it
+        if (independent_disabled == NETDATA_ZSWAP_SITE_END)
+            return -1;
+    }
```

Comment:
Why `if (independent_disabled == NETDATA_ZSWAP_SITE_END)` is inside `else`?

Reply:
If we cannot collect grouped metrics we test the independent. 
I remember when we were working together with ancient kernel(4.4.0) there was only one metric not collected.
Please, confirm another thing. I observed that I have metrics with zero values when `zram` is disabled, can you confirm that you also have them?

Reply:
I simplified the main loop, now the module exits when 0 metrics are collected.

Also, let's call main functions like `do_*` as in the proc plugin.